SteamKit2/SteamKit2/Steam/Handlers/SteamUserStats/Callbacks.cs

```diff
@@ -33,5 +36,112 @@ internal NumberOfPlayersCallback( JobID jobID, MsgClientGetNumberOfCurrentPlayer
+        {
+            /// <summary>
+            /// Gets the result of the request.
+            /// </summary>
+            public EResult Result { get; private set; }
+            /// <summary>
+            /// Leaderboard ID.
+            /// </summary>
+            public int ID { get; private set; }
+            /// <summary>
+            /// How many entires there are for requested leaderboard.
+            /// </summary>
+            public int EntryCount { get; private set; }
+            /// <summary>
+            /// Sort method to use for this leaderboard.
+            /// </summary>
+            public ELeaderboardSortMethod SortMethod { get; private set; }
+            /// <summary>
+            /// Display type for this leaderboard.
+            /// </summary>
+            public ELeaderboardDisplayType DisplayType { get; private set; }
+
+
+            internal FindOrCreateLeaderboardCallback( JobID jobID, CMsgClientLBSFindOrCreateLBResponse resp )
+            {
+                this.JobID = jobID;
+
+                this.Result = ( EResult )resp.eresult;
+                this.ID = resp.leaderboard_id;
+                this.EntryCount = resp.leaderboard_entry_count;
+                this.SortMethod = ( ELeaderboardSortMethod )resp.leaderboard_sort_method;
+                this.DisplayType = ( ELeaderboardDisplayType )resp.leaderboard_display_type;
+            }
+        }
+
+        /// <summary>
+        /// This callback is fired in response to <see cref="GetLeaderboardEntries" />.
+        /// </summary>
+        public class LeaderboardEntriesCallback : CallbackMsg
+        {
+            /// <summary>
+            /// Represents a single package in this response.
+            /// </summary>
+            public sealed class LeaderboardEntry
+            {
+                /// <summary>
+                /// Gets the <see cref="SteamID"/> for this entry.
+                /// </summary>
+                public SteamID SteamID { get; private set; }
+                /// <summary>
+                /// Gets the global rank for this entry.
+                /// </summary>
+                public int GlobalRank { get; private set; }
+                /// <summary>
+                /// Gets the score for this entry.
+                /// </summary>
+                public int Score { get; private set; }
+                /// <summary>
+                /// Gets the <see cref="UGCHandle"/> attached to this entry.
+                /// </summary>
+                public UGCHandle UGCId { get; private set; }
+
+
+                internal LeaderboardEntry( CMsgClientLBSGetLBEntriesResponse.Entry entry )
+                {
+                    GlobalRank = entry.global_rank;
+                    Score = entry.score;
+                    SteamID = new SteamID( entry.steam_id_user );
+                    UGCId = new UGCHandle( entry.ugc_id );
+
+                    // entry.details // TODO: ???
```

Comment:
We'd probably want to expose this the same way Steam does, as an array of 32 bit ints.